python/pyspark/testing/connectutils.py

```diff
@@ -47,6 +49,9 @@ def __init__(self) -> None:
     def set_hook(self, name: str, hook: Any) -> None:
         self.hooks[name] = hook
 
+    def drop_hook(self, name: str) -> None:
+        self.hooks.pop(name)
```

Comment:
should we catch the exceptions if `self.hooks` do not contains `name`?

Reply:
This is only for our testing and we have good control over it (e.g. make sure we only clean up things that registered for the testing) so I think it is ok.